packages/frontend/src/components/ReactHookForm/AutoComplete.tsx

```diff
@@ -84,14 +133,65 @@ const ControlledSuggest: FC<{
         string | `${string}.${string}` | `${string}.${number}`
     >;
     suggestProps?: Partial<React.ComponentProps<typeof Suggest2<Item>>>;
-}> = ({ isLoading, suggestProps, disabled, items, field, ...props }) => {
+    groupBy?: (item: Item) => string;
+}> = ({
+    isLoading,
+    suggestProps,
+    disabled,
+    items,
+    field,
+    groupBy,
+    ...props
+}) => {
     const activeItem = items.find((item) => item.value === field.value);
     const onItemSelect = useCallback(
         (item: Item) => {
             field.onChange(item.value);
         },
         [field],
     );
+
+    const renderGroupedItemList = useMemo(() => {
+        const itemListRenderer: ItemListRenderer<Item> = (
+            listProps: ItemListRendererProps<Item>,
+        ) => {
+            const initialContent = items;
+            const noResults = (
+                <MenuItem
+                    disabled={true}
+                    text="No results."
+                    roleStructure="listoption"
+                />
+            );
+
+            // omit noResults if createNewItemFromQuery and createNewItemRenderer are both supplied, and query is not empty
+            const createItemView = listProps.renderCreateItem();
+            const maybeNoResults = createItemView != null ? null : noResults;
+
+            const menuContent = renderGroupedMenuContent(
+                listProps,
+                maybeNoResults,
+                groupBy,
+            );
+            if (menuContent == null && createItemView == null) {
+                return null;
+            }
+            const { createFirst } = { createFirst: undefined }; // this.state;
```

Comment:
I don't understand createItemView and what is happening here.
but I might have a suggestion here. since this is AutoComplete component which is quite common.. maybe we should just expose itemListRenderer prop outside and then provide rendering function from the outside and customise depending on the requirements of the feature.

Reply:
I've copied most of the new methods from here https://github.com/palantir/blueprint/blob/develop/packages/docs-app/src/examples/select-examples/selectExample.tsx 

I've exposed the "groupBy" method, which is the only thing you want to change if you want to use grouping. Everything else is boilerplate and Item /Menu properties. But IF don't mind moving it outside, what do you think ?

Reply:
I see. then I only have question why we need
this line
```const { createFirst } = { createFirst: undefined }; // this.state;```

Reply:
I've removed all the configuration we don't need from the methods .